src/qibolab/instruments/zhinst.py

```diff
@@ -0,0 +1,997 @@
+                    with exp.match_local(handle=f"acquire{qubit.name}"):
+                        with exp.case(state=0):
+                            pass
+                        with exp.case(state=1):
+                            exp.play(signal=f"drive{qubit.name}", pulse=ZhPulse(fast_reset[qubit.name]).zhpulse)
+
+    def sweep(self, qubits, sequence, *sweepers, options):
+        """Play pulse and sweepers sequence"""
+        if options.relaxation_time is None:
+            options.relaxation_time = self.relaxation_time
+
+        self.sweepers = list(sweepers)
+        self.sequence_zh(sequence, qubits, sweepers)
+        self.calibration_step(qubits)
+        self.create_exp(qubits, options)
+        self.run_exp()
+
+        # TODO: Test for several readouts and qubits simultaneosuly
+        results = {}
+        for qubit in qubits.values():
+            if not qubit.flux_coupler:
+                if self.sequence[f"readout{qubit.name}"]:
+                    exp_res = self.results.get_data(f"sequence{qubit.name}")
+                    i = exp_res.real
+                    q = exp_res.imag
+                    results[self.sequence[f"readout{qubit.name}"][0].pulse.serial] = ExecutionResults.from_components(
+                        i, q
+                    )
+
+        # self.run_sim(sim_time) #FIXME: Careful placement or reconnection to avoid messing with several executions
+        # lo.show_pulse_sheet("pulses", self.exp) #FIXME: Wait until next LO release for bugs
+        # self.disconnect() #FIXME: Careful placement or reconnection to avoid messing with several executions
+        return results
+
+    # TODO: Recursion tests and Better sweeps logic
+    def sweep_recursion(self, qubits, exp, exp_calib, relaxation_time, acquisition_type, fast_reset):
+        """Sweepers recursion for multiple nested sweepers"""
+        # Ordered like this for how they defined the frequncy sweep on qibocal to be the last
+        # and I need it to be the outer one
+        sweeper = self.sweepers[-1]
+        i = len(self.sweepers) - 1
+        self.sweepers.remove(sweeper)
+
+        parameter = None
+        print("sweep", sweeper.parameter.name.lower())
+
+        if sweeper.parameter.name.lower() == "frequency":
+            for pulse in sweeper.pulses:
+                qubit = pulse.qubit
+                if pulse.type.name.lower() == "drive":
+                    line = "drive"
+                elif pulse.type.name.lower() == "readout":
+                    line = "measure"
+                exp_calib[f"{line}{qubit}"] = lo.SignalCalibration(
+                    oscillator=lo.Oscillator(
+                        frequency=ZhSweeper(pulse, sweeper, qubits[sweeper.pulses[0].qubit]).zhsweeper,
+                        modulation_type=lo.ModulationType.HARDWARE,
+                    )
+                )
+
+        if sweeper.parameter.name == "bias":
```

Comment:
You can use `elif`
```suggestion
        elif sweeper.parameter is Parameter.bias:
```
here and in the other `if`s that follow so that you only execute one of them. Only one will be true anyway.

Another possibility is to create different small functions for each case and have a dispatcher, like I did in a revision for QM here
https://github.com/qiboteam/qibolab/blob/d5b6a0248e7257488bb5d320c69f95495a3df540/src/qibolab/instruments/qm.py#L772

but in your case this may be an "overkill" since what happens under every `if` is not very complicated anyway.